Mamba information and install instructions (#156)
mila-iqia/mila-docs: docs/Userguide_python.rst

.. prompt:: bash (<env>)$

   conda clean --all

**Mamba**

When installing new packages with ``conda install``, conda uses a built-in 
dependency solver for solving the dependency graph of all packages (and their 
versions) requested such that package dependency conflicts are avoided.

In some cases, especially when there are many packages already installed in a 
conda environment, conda's built-in dependency solver can struggle to solve the 
dependency graph, taking several to tens of minutes, and sometimes never 

> solving. In these cases, it is recommended to try `libmamba 

Reviewer: ```suggestion
solving. In these cases, you can try the alternative solver `libmamba 
```

Reviewer: If @obilaniu gives his seal of approval we could keep the previous form but not so long ago I think he had some reserve on the mass suggestion of mamba. this could have changed if mamba is now stable

Reviewer 2: @satyaog We've discussed this in the original conversation, but in summary Mamba doesn't produce identical solves to Conda. Still, it has become more robust, and it's become the only practical solution in some cases.

As of the newest PyTorch 1.13.x's install instructions, any deviation from `-c pytorch -c nvidia` (such as by adding the incredibly common `-c conda-forge`) results in `conda create`/`conda install` taking _ridiculous_ amounts of time to come up with a solution. `mamba` on the other hand has no problems.

Definitely go for merge.

<https://conda.github.io/conda-libmamba-solver/getting-started/>`_.

To install and set the ``libmamba`` solver, run the following commands:

.. prompt:: bash $, auto

>    # Install miniconda
>    # (you can not use the preinstalled anaconda/miniconda as installing libmamba
>    #  requires ownership over the anaconda/miniconda install directory)
>    $ wget https://repo.anaconda.com/miniconda/Miniconda3-py310_22.11.1-1-Linux-x86_64.sh
>    $ bash Miniconda3-py310_22.11.1-1-Linux-x86_64.sh
> 
>    # Install libmamba
>    $ conda install -n base conda-libmamba-solver

Reviewer: ```suggestion
   \# Install miniconda
   \# (you can not use the preinstalled anaconda/miniconda as installing libmamba
   \#  requires ownership over the anaconda/miniconda install directory)
   $ wget https://repo.anaconda.com/miniconda/Miniconda3-py310_22.11.1-1-Linux-x86_64.sh
   $ bash Miniconda3-py310_22.11.1-1-Linux-x86_64.sh

   \# Install libmamba
   $ conda install -n base conda-libmamba-solver
```

>    \# Install miniconda
>    \# (can not use anaconda/miniconda not installed by you as installing libmamba 
>    \#  requires ownership over the anaconda/miniconda install directory)
>    $ wget https://repo.anaconda.com/miniconda/Miniconda3-py310_22.11.1-1-Linux-x86_64.sh
>    $ bash Miniconda3-py310_22.11.1-1-Linux-x86_64.sh
> 
>    \# Install libmamba
>    $ conda install -n base conda-libmamba-solver

Reviewer: ```suggestion
   # Install miniconda
   # (you can not use the preinstalled anaconda/miniconda as installing libmamba
   #  requires ownership over the anaconda/miniconda install directory)
   $ wget https://repo.anaconda.com/miniconda/Miniconda3-py310_22.11.1-1-Linux-x86_64.sh
   $ bash Miniconda3-py310_22.11.1-1-Linux-x86_64.sh

   # Install libmamba
   $ conda install -n base conda-libmamba-solver
```

Reviewer: Sorry about suggesting the removal of the `\#` to `#` only. This looks bad so I reverted
